### PR TITLE
Support fail-fast label

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/10006?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10006/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10006
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Add a pytest runner script that optionally uses `--exitfirst` flag when the PR is marked as `fail-fast`.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: dev/pytest.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+
+def main() -> None:
+    fail_fast = "fail-fast" in fetch_labels()
+    extra_options = ["--exitfirst"] if fail_fast else []
+    pytest.main([*extra_options, *sys.argv[1:]])
+
+
+if __name__ == "__main__":
+    main()
```
Comment (harupy) [inline dev/pytest.py:35]: /ai translate this code to a shell script. jq can be used if necessary.

File: tests/evaluate/test_default_evaluator.py [hunk 2]
```diff
@@ -286,6 +286,7 @@ def test_multi_classifier_evaluation(
     use_sample_weights,
 ):
     sample_weights = np.random.rand(len(iris_dataset.labels_data)) if use_sample_weights else None
+    assert False, "test"
```
Comment (harupy) [inline tests/evaluate/test_default_evaluator.py:289]:
```suggestion
```

File: mlflow/ml-package-versions.yml [hunk 3]
```diff
@@ -126,7 +126,7 @@ xgboost:
       ">= 0.0.0": ["scikit-learn"]
       "< 1.6": ["pandas<2"]
     run: |
-      pytest tests/xgboost/test_xgboost_model_export.py
+      dev/pytest.sh tests/xgboost/test_xgboost_model_export.py
```
Comment (harupy) [inline mlflow/ml-package-versions.yml:129]: We have more pytest invocations. I'll wait for a day to ensure this change breaks nothing.

File: tests/evaluate/test_default_evaluator.py (hunk 2 above)
Comment (harupy) [reply tests/evaluate/test_default_evaluator.py:289]: https://github.com/mlflow/mlflow/actions/runs/6569477321/job/17845428508?pr=10006

**Review by daniellok-db — APPROVED** [review]
lgtm, i agree with waiting a day as well

**Review by serena-ruan — APPROVED** [review]
LGTM!

**Pull request merged**